doc/src/sgml/monitoring.sgml

```diff
@@ -4522,6 +4522,200 @@ description | Waiting for a newly initialized WAL file to reach durable storage
+       not been freed are considered long lived and will continue to be part of
+       <structfield>global_dsm_allocated_bytes</structfield>, found in the
+       <link linkend="monitoring-pg-stat-global-memory-allocation-view">
+       <structname>pg_stat_global_memory_allocation</structname></link> view.
+      </para></entry>
+     </row>
+
+     <row>
+      <entry role="catalog_table_entry"><para role="column_definition">
+       <structfield>generation_allocated_bytes</structfield> <type>bigint</type>
+      </para>
+      <para>
+       Subtotal of memory allocated via the generation allocator.
+      </para></entry>
+     </row>
+
+     <row>
+      <entry role="catalog_table_entry"><para role="column_definition">
+       <structfield>slab_allocated_bytes</structfield> <type>bigint</type>
+      </para>
+      <para>
+       Memory currently allocated to this backend via the slab allocator.
+      </para></entry>
+     </row>
+
+    </tbody>
+   </tgroup>
+  </table>
+
+ </sect2>
+
+ <sect2 id="monitoring-pg-stat-global-memory-allocation-view">
+  <title><structname>pg_stat_global_memory_allocation</structname></title>
+
+  <indexterm>
+   <primary>pg_stat_global-memory_allocation</primary>
+  </indexterm>
+
+  <para>
+   The <structname>pg_stat_global_memory_allocation</structname> view shows
+   the current memory allocation for the entire server. This view
+   has a single row with a column for each global value.
+   Due to the dynamic nature of memory allocations, the
+   values may not be exact, but they should be sufficient for monitoring the server.
+   Use <function>pg_size_pretty</function> described in
+   <xref linkend="functions-admin-dbsize"/> to make the byte populated values
+   more easily readable.
+  </para>
+
+  <table id="pg-stat-global-memory-allocation-view" xreflabel="pg_stat_global_memory_allocation">
+   <title><structname>pg_stat_global_memory_allocation</structname> View</title>
+   <tgroup cols="1">
+    <thead>
+     <row>
+      <entry role="catalog_table_entry"><para role="column_definition">
+       Column Type
+      </para>
+      <para>
+       Description
+      </para></entry>
+     </row>
+    </thead>
+
+    <tbody>
+     <row>
+      <entry role="catalog_table_entry"><para role="column_definition">
+       <structfield>datid</structfield> <type>oid</type>
+      </para>
+      <para>
+       OID of the database this backend is connected to
+      </para></entry>
+     </row>
+
+     <row>
+      <entry role="catalog_table_entry"><para role="column_definition">
+       <structfield>total_memory_used</structfield> <type>bigint</type>
+      </para>
+     <para>
+      Reports the total amount of memory (in bytes) currently used by the server.
+      This total includes shared memory segments, actual memory
+      allocated by the server, along with a fixed amount of memory
+      assumed for each server process. Due to the dynamic nature of memory allocation,
```

Comment:
This in't just a fixed amount of memory but is actually how much memory we are currently tracking for each backend, right..?  Plus maybe 1MB that we kind of assume is what's needed per-backend that just gets copied as part of being modified post-fork, but I don't think I'd even mention that here- it's covered in the "this isn't exact" comment, imv.  Certainly part of the idea here is that folks can monitor this value vs. the max-total-memory value to see if they're getting close to running into that limit and having queries start failing...